src/ServiceManagement/Common/Commands.ScenarioTest/Resources/ChefExtension/ChefExtensionTests.ps1

```diff
@@ -0,0 +1,95 @@
+
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+
+#
+# ChefExtensionTests.ps1
+#
+
+$PLACEHOLDER = "PLACEHOLDER1@";
+
+<#
+.SYNOPSIS
+Test the usage of the Set virtual machine chef extension command
+#>
+
+# Setup
+$vmName = "vmtwo"
+$svcName = "storone"
+$storageName = "storone"
+$location = "West US"
+
+function Test-SetAzureVMChefExtension
+{	
+    try
+	{
+		New-AzureStorageAccount -StorageAccountName $storageName -Location $location
+        Set-CurrentStorageAccountName $storageName
+
+		New-AzureService -ServiceName $svcName -Location $location
+		New-AzureQuickVM -Windows -ImageName "a699494373c04fc0bc8f2bb1389d6106__Windows-Server-2012-R2-BYOL-20160915-en.us-127GB.vhd" -Name $vmName -ServiceName $svcName -AdminUsername "pstestuser" -Password $PLACEHOLDER
+
+		$vm = Get-AzureVM -ServiceName $svcName -Name $vmName
+
+		Set-AzureVMChefExtension -VM $vm -ValidationPem ".\Resources\ChefExtension\tstorgnztn-validator.pem" -ClientRb ".\Resources\ChefExtension\client.rb" -JsonAttribute '{"container_service": {"chef-init-test": {"command": "C:\\opscode\\chef\\bin"}}}' -ChefServiceInterval 35 -Windows
```

Comment:
The issue with your tests is that, when runnign tests in parallel, the current directory can change.  The best option to fix this is to grab the value of AppDomain.Current.BaseDirectory, and pass this to your script, so you are not dependent on the current directory